autopilot/contractor/hosts_test.go

```diff
@@ -52,20 +52,27 @@ func TestScoredHostsRandSelectByScore(t *testing.T) {
 		t.Fatal("unexpected")
 	}
 
-	// assert select is random on equal inputs
-	counts := make([]int, 2)
+	// assert select is random on equal inputs, we calculate the chi-square
+	// statistic and assert it's less than critical value of 6.635 (1 degree of
+	// freedom, using alpha of .01)
+	var counts [2]int
 	hosts = scoredHosts{
 		{score: .1, host: api.Host{PublicKey: types.PublicKey{1}}},
 		{score: .1, host: api.Host{PublicKey: types.PublicKey{2}}},
 	}
-	for i := 0; i < 100; i++ {
+	nRuns := 1e5
+	for i := 0; i < int(nRuns); i++ {
 		if hosts.randSelectByScore(1)[0].host.PublicKey == (types.PublicKey{1}) {
 			counts[0]++
 		} else {
 			counts[1]++
 		}
 	}
-	if diff := absDiffInt(counts[0], counts[1]); diff > 40 {
-		t.Fatal("unexpected", counts[0], counts[1], diff)
+	var chi2 float64
+	for i := 0; i < 2; i++ {
+		chi2 += math.Pow(float64(counts[i])-nRuns/2, 2) / nRuns / 2
```

Comment:
If the expected value is `nRuns/2` shouldn't this be `math.Pow(float64(counts[i])-nRuns/2, 2) / (nRuns / 2)`?
Should make a difference whether you divided by `10000` and then `2` or `5000`

Reply:
lol thanks, I blame the chad